### PR TITLE
refactor!: remove outdated widget properties and document used widget properties

### PR DESCRIPTION
**Pull request by flekschas**
**BREAKING CHANGE:** This PR removes several outdated and unused helper widget properties, i.e., widget properties that create helper widgets. It additionally adds documentation for the widget properties that define the view model.

## Description

> What was changed in this pull request?

The following list of helper widgets for configuring the scatter are removed from the `scatter.widget` as they are unmaintained, undocumented, and incomplete. If you relied on any of those UI widgets, please see `jscatter/widget.py` v0.15.0 on how they were created.

- `scatter.widget.mouse_mode_widget`
- `scatter.widget.lasso_initiator_widget`
- `scatter.widget.selection_widget`
- `scatter.widget.hovering_widget`
- `scatter.widget.color_widgets`
- `scatter.widget.color_by_widget`
- `scatter.widget.color_map_widget`
- `scatter.widget.height_widget`
- `scatter.widget.background_color_widget`
- `scatter.widget.background_image_widget`
- `scatter.widget.lasso_color_widget`
- `scatter.widget.lasso_min_delay_widget`
- `scatter.widget.lasso_min_dist_widget`
- `scatter.widget.color_widget`
- `scatter.widget.color_selected_widget`
- `scatter.widget.color_hover_widget`
- `scatter.widget.opacity_widget`
- `scatter.widget.selection_outline_width_widget`
- `scatter.widget.size_widget`
- `scatter.widget.selection_size_addition_widget`
- `scatter.widget.reticle_widget`
- `scatter.widget.reticle_color_widget`
- `scatter.widget.download_view_widget`
- `scatter.widget.save_view_widget`
- `scatter.widget.reset_view_widget`

Additionally, the following helper methods are removed as they are unnecessary.

- `scatter.widget.options()` (simply listed out all above removed widgets)
- `scatter.widget.select()` (same as `scatter.selection = list_of_point_indices`)
- `scatter.widget.use_cmap()` (same as passing the cmap name to `scatter.color(map=cmap_name)`)

> Why is it necessary?

The widget properties were initially added for fast prototyping but have not been updated or documented since the very early days. Maintaining those outdated properties is not 

## Checklist

- [x] Provided a concise title as a [semantic commit message](https://www.conventionalcommits.org) (e.g. "fix: correctly handle undefined properties")
- [x] `CHANGELOG.md` updated
- [ ] Tests added or updated
- [x] Documentation in `API.md`/`README.md` added or updated
- [ ] Example(s) added or updated
- [ ] Screenshot, gif, or video attached for visual changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,45 @@
+## v0.16.1
```
Comment (manzt) [inline CHANGELOG.md:1]: Would it make sense to just cut a `0.17.0` instead of a breaking patch release?

File: docs/api.md [hunk 2]
```diff
@@ -639,8 +639,118 @@ You can define those property when creating a `Scatter` instance. For example,
 
 ### Widget
 
-_So sorry, I haven't had a chance to document the widget properties yet._
-
+The widget has the following properties, which you can think of as the view model of Jupyter Scatter.
+
+::: warning
+While you can adjust these properties directly, we highly recommend using the [`Scatter` methods](#methods) as these methods purely exist to simplify and automate the setting of the widget properties.
```
Comment (manzt) [inline docs/api.md:645]:
```suggestion
While you can adjust these properties directly, the [`Scatter` methods](#methods) are the idiomatic and recommended way to set widget properties.
```

**Review by manzt — APPROVED** [review]
Looks good to me. Just a note on the version to push. I'm at the scverse hackathon this week and could maybe have a stab at setting up an initial pytest config here if that would be welcomed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,45 @@
+## v0.16.0
```
Comment (manzt) [inline CHANGELOG.md:1]: Just flagging to remember v0.17.0

Comment (flekschas) [reply CHANGELOG.md:1]: The current version is 0.15 so jumping to 0.16 should be sufficient

Comment (manzt) [reply CHANGELOG.md:1]: woops, sorry!

**Pull request merged**